scenes/game/local_game.gd

```diff
@@ -4318,6 +4333,94 @@ func handle_strike_effect(card_id :int, effect, performing_player : Player):
+						"effect_type": "opponent_discard_range_or_reveal",
+						"target_range": decision_info.valid_zones[i],
+						"amount": 1
+					})
+
+				change_game_state(Enums.GameState.GameState_PlayerDecision)
+				events += [create_event(Enums.EventType.EventType_PickNumberFromRange, performing_player.my_id, 0)]
+			else:
+				assert(false, "Target effect for name_range not found.")
+				decision_info.clear()
+		"opponent_discard_range_or_reveal":
+			var target_range = effect['target_range']
+			var card_ids_in_range = []
+			if target_range is String:
+				if target_range == "Range X":
+					# If the range is a string like "TOTAL_POWER".
+					for card in opposing_player.hand:
+						if card.definition['range_min'] is String or card.definition['range_max'] is String:
+							card_ids_in_range.append(card.id)
+				elif target_range == "Range -":
+					# If the range is -1 like Block.
+					for card in opposing_player.hand:
+						var card_range = card.definition['range_min']
+						if is_number(card_range) and card_range == -1:
+							card_ids_in_range.append(card.id)
+				else:
+					assert(false, "Unknown target range")
+			else:
+				# If the range is an actual number.
+				for card in opposing_player.hand:
+					var card_range_min = card.definition['range_min']
+					var card_range_max = card.definition['range_max']
+					# Some cards have mixed numbers and strings.
+					# Potentially evaluate? Currently only Phonon though and max==min if not in strike.
+					if is_number(card_range_min) and is_number(card_range_max):
+						if target_range >= card_range_min and target_range <= card_range_max:
+							card_ids_in_range.append(card.id)
+					elif is_number(card_range_min) and target_range == card_range_min:
+						card_ids_in_range.append(card.id)
+					elif is_number(card_range_max) and target_range == card_range_max:
+						card_ids_in_range.append(card.id)
+			if card_ids_in_range.size() > 0:
+				# Opponent must choose one of these cards to discard.
+				var amount = effect['amount']
+				change_game_state(Enums.GameState.GameState_PlayerDecision)
+				decision_info.clear()
+				decision_info.type = Enums.DecisionType.DecisionType_ChooseToDiscard
+				decision_info.effect_type = "opponent_discard_choose_internal"
+				decision_info.effect = effect
+				decision_info.bonus_effect = null
+				decision_info.destination = "discard"
+				decision_info.limitation = "from_array"
+				if target_range is String:
+					decision_info.extra_info = "include %s" % target_range
+				else:
+					decision_info.extra_info = "include Range %s" % target_range
+				decision_info.choice = card_ids_in_range
+				decision_info.can_pass = false
+
+				decision_info.choice_card_id = card_id
+				decision_info.player = opposing_player.my_id
+				events += [create_event(Enums.EventType.EventType_Strike_ChooseToDiscard, opposing_player.my_id, amount)]
+			else:
+				# Didn't have any that matched, so forced to reveal hand.
+				events += opposing_player.reveal_hand()
```

Comment:
Maybe a small log message here would be nice (and also one mentioning the range that the player named, for both cases)